include/ginkgo/core/preconditioner/isai.hpp

```diff
@@ -211,15 +233,21 @@ class Isai : public EnableLinOp<Isai<IsaiType, ValueType, IndexType>>,
         : EnableLinOp<Isai>(factory->get_executor(), system_matrix->get_size()),
           parameters_{factory->get_parameters()}
     {
+        const auto exec = this->get_executor();
         const auto skip_sorting = parameters_.skip_sorting;
         const auto power = parameters_.sparsity_power;
         const auto excess_limit = parameters_.excess_limit;
         generate_inverse(system_matrix, skip_sorting, power, excess_limit);
+
+        auto csr_inv = as<Csr>(approximate_inverse_);
+        auto ell_inv = convert_csr_to_ell(approximate_inverse_.get());
```

Comment:
One thing to note is that `generate_inverse` creates a `Csr<ValueType, IndexType>` but `convert_csr_to_ell()` takes in a `Csr<StorageType, IndexType>`? That makes the functions `convert_csr_to_ell()` and `convert_ell_to_csr()` non symmetric and it's probably unneeded as well? It's the same thing in other places as well.

Reply:
Actually, `convert_csr_to_ell()` takes in a `Csr<ValueType, IndexType>` and converts it to an `Ell<StorageType, IndexType>`. `convert_ell_to_csr()` in turn converts an `Ell<StorageType, IndexType>` to a `Csr<ValueType, IndexType>`.

Reply:
Ah right, I missed some lines. Thanks for the explanation.